router.js

```diff
@@ -0,0 +1,47 @@
+module.exports = require('injectdeps')(['container', 'logger'], function(container, logger) {
+  return (request, response, next) => {
+    let methodAndPath = `${request.method} ${request.path}`;
+    // First check if the path is supported by our swagger definition
+    // Note that we're assuming that another module has checked this and decorated the request object
+    // The basic implementation is the swagger-tools/metadata middleware exposed through the tools module
+    if(!request.swagger || !request.swagger.operation) {
+      response.statusCode = 404;
+      return next(new Error(`Unsupported endpoint ${methodAndPath}`));
+    }
+
+    let controllerName = request.swagger['x-swagger-router-controller'];
+
+    if(!controllerName) {
+      return next(new Error(`Swagger specification does not specify a controller for ${methodAndPath}`));
+    }
+
+    let fullControllerName = 'controller.' + controllerName;
```

Comment:
I'm not particularly keen on hard coding this. I'll make this an option that can be passed in